recipes/pulseaudio/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+            self.requires("xorg/system")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1h")   
+        # FIXME: enable when #2147 is merged
+        # if self.options.with_dbus
+        #     self.requires("dbus/1.12.16")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            self._autotools = AutoToolsBuildEnvironment(self)
+            args=[]
+            
+            # FIXME: add dbus when #2147 is merged
+            for lib in ["alsa", "x11", "openssl"]:
+                args.append("--%s-%s" % ("enable" if getattr(self.options, "with_" + lib) else "disable", lib))
+            args.append("--%s-glib2" % ("enable" if self.options.with_glib else "disable"))
+            args.append("--%s-fftw" % ("with" if self.options.with_fftw else "without"))
+            if self.options.shared:
+                args.extend(["--enable-shared=yes", "--enable-static=no"])
+            else:
+                args.extend(["--enable-shared=no", "--enable-static=yes"])
+            args.append("--with-udev-rules-dir=%s" % os.path.join(self.package_folder, "bin", "udev", "rules.d"))
+            with tools.environment_append({"PKG_CONFIG_PATH": self.build_folder}):
+                with tools.environment_append({
+                        "FFTW_CFLAGS": tools.PkgConfig("fftw").cflags,
+                        "FFTW_LIBS": tools.PkgConfig("fftw").libs} if self.options.with_fftw else tools.no_op()):
+                    with tools.environment_append(RunEnvironment(self).vars):
+                        self._autotools.configure(args=args,  configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "etc"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        for f in glob.glob(os.path.join(self.package_folder, "lib", "**", "*.la"), recursive=True):
+            os.remove(f)
+
+    def package_info(self):
+        self.cpp_info.libdirs = ["lib", os.path.join("lib", "pulseaudio")]
+        if self.options.with_glib:
+            self.cpp_info.libs.append("pulse-mainloop-glib")
+        self.cpp_info.libs.extend(["pulse-simple", "pulse"])
+        if not self.options.shared:
+            self.cpp_info.libs.append("pulsecommon-%s" % self.version)
+        self.cpp_info.defines = ["_REENTRANT"]
```

Comment:
I think this is a define for the C runtime, not for pulseaudio.
Is it actually used in the pulse headers?

Reply:
I don't know much, but this define is present in upstream .pc file

Reply:
I've just built it on my machine and it is indeed present in the `.pc` file.
A quick grep came up empty.

See https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html#Feature-Test-Macros
These are obsolete defines.

Reply:
But I guess they are harmless..

Reply:
I think it's better if we are conform to the upstream project. For some reason, some users could rely on this macro to be defined. In other word we should submit the change upstream